google/cloud/documentai_toolbox/wrappers/hocr.py

```diff
@@ -0,0 +1,221 @@
+        start_index = line.layout.text_anchor.text_segments[0].start_index
+        end_index = line.layout.text_anchor.text_segments[0].end_index
+        words = [word for word in page.tokens if word.layout.text_anchor.text_segments[0].start_index >= start_index if word.layout.text_anchor.text_segments[0].end_index <= end_index]
+        l.append(_HocrLine(words=words, text=(_get_text(line, document_text)), documentai_line=line, page=page, dimensions=(page.dimension), document_text=document_text))
+    else:
+        return l
+
+
+def _load_hocr_paragraphs(paragraphs: List[documentai.Document.Page.Paragraph], page: documentai.Document.Page, document_text: str):
+    p = []
+    for paragraph in paragraphs:
+        start_index = paragraph.layout.text_anchor.text_segments[0].start_index
+        end_index = paragraph.layout.text_anchor.text_segments[0].end_index
+        lines = [line for line in page.lines if line.layout.text_anchor.text_segments[0].start_index >= start_index if line.layout.text_anchor.text_segments[0].end_index <= end_index]
+        p.append(_HocrParagraph(lines=lines, page=page, dimensions=(page.dimension), document_text=document_text, documentai_paragraph=paragraph))
+    else:
+        return p
+
+
+def _load_hocr_blocks(page: documentai.Document.Page, document_text: str):
+    blocks = []
+    for block in page.blocks:
+        start_index = block.layout.text_anchor.text_segments[0].start_index
+        end_index = block.layout.text_anchor.text_segments[0].end_index
+        paragraphs = [p for p in page.paragraphs if p.layout.text_anchor.text_segments[0].start_index >= start_index if p.layout.text_anchor.text_segments[0].end_index <= end_index]
+        blocks.append(_HocrBlock(paragraphs=paragraphs, page=page, dimensions=(page.dimension), document_text=document_text, documentai_block=block))
+    else:
+        return blocks
+
+
+def _load_pages(pages: List[documentai.Document.Page], document_text: str, filename: str):
+    res_pages = []
+    for page in pages:
+        res_pages.append(_HocrPage(documentai_page=page.documentai_page, document_text=document_text, filename=filename))
+
+    return res_pages
+
+
+@dataclasses.dataclass
+class _HocrWord:
+    text: str = dataclasses.field(repr=False)
+    documentai_word: documentai.Document.Page.Token = dataclasses.field(repr=False)
+    dimensions: documentai.Document.Page.Dimension = dataclasses.field(repr=False)
+    bounding_box: str = dataclasses.field(init=False, repr=False)
+
+    def __post_init__(self):
+        self.bounding_box = _get_bounding_box(element_with_layout=(self.documentai_word), dimensions=(self.dimensions))
+
+    def to_hocr(self, pidx: int,bidx: int, paridx: int, lidx: int, widx: int):
+        f = ""
+        word_text = self.text.replace("&","&amp;").replace("<","&lt;").replace(">","&gt;")
+        f += f"<span class='ocrx_word' id='word_{pidx}_{bidx}_{paridx}_{lidx}_{widx}' title='{self.bounding_box}'>{word_text}</span>\n"       
```

Comment:
Similarly, there are standard libraries for manipulating and creating XML: https://docs.python.org/3/library/xml.etree.elementtree.html

Reply:
oh i did not know i could do this through a library i'll take a look.